src/functionalTest/kotlin/com/github/jengelman/gradle/plugins/shadow/transformers/AppendingTransformerTest.kt

```diff
@@ -100,6 +104,31 @@ class AppendingTransformerTest : BaseTransformerTest() {
     )
   }
 
+  @Test
+  fun duplicatesStrategyFailShouldFailBuildForDuplicatedEntries() {
+    writeMainClass()
+    projectScriptPath.appendText(
+      """
+        dependencies {
+          ${implementationFiles(buildJarOne(), buildJarTwo())}
+        }
+        $shadowJar {
+          // This should fail the build because of the duplicates.
+          duplicatesStrategy = DuplicatesStrategy.FAIL
+        }
+      """.trimIndent(),
+    )
+
+    val result = runWithFailure(shadowJarTask)
+
+    assertThat(result).all {
+      taskOutcomeEquals(shadowJarTask, FAILED)
```

Comment:
This works for a custom fat jar, see 388909da07b900df1f6c9b2e3889b800611950f9.

Reply:
As `ZipTree` inputs are handled by https://github.com/gradle/gradle/blob/df5bc230c57db70aa3f6909403e5f89d7efde531/platforms/core-configuration/file-operations/src/main/java/org/gradle/api/internal/file/copy/DuplicateHandlingCopyActionDecorator.java#L54-L65